tests/sealKey/enc/enc.cpp

```diff
@@ -247,37 +188,29 @@ bool TestOEGetSealKey()
             return false;
         }
 
-        oe_free_seal_key(second_key_buffer_ptr, key_info_ptr);
+        oe_free_seal_key(second_key_buffer_ptr, NULL);
 
         OE_TEST(second_key_buffer_ptr_size == sizeof(sgx_key_t));
         OE_TEST(key_info_ptr_size == sizeof(sgx_key_request_t));
 
-        uint8_t third_key_buffer[sizeof(sgx_key_t)] = {0};
         uint8_t* third_key_ptr;
-        size_t third_key_buffer_size = sizeof(third_key_buffer);
         size_t third_key_ptr_size;
 
         // Get the seal key using saved key info.
-        ret = oe_get_seal_key_v1(
-            key_info, key_info_size, third_key_buffer, &third_key_buffer_size);
-        if (ret != OE_OK)
-        {
-            return false;
-        }
         ret = oe_get_seal_key_v2(
-            key_info, key_info_size, &third_key_ptr, &third_key_ptr_size);
+            key_info_ptr,
+            key_info_ptr_size,
+            &third_key_ptr,
+            &third_key_ptr_size);
         if (ret != OE_OK)
         {
             return false;
         }
 
-        OE_TEST(third_key_buffer_size == sizeof(sgx_key_t));
         OE_TEST(third_key_ptr_size == sizeof(sgx_key_t));
 
         // The seal keys should match.
-        if ((memcmp(key_buffer, second_key_buffer, sizeof(sgx_key_t)) != 0) ||
-            (memcmp(key_buffer, third_key_ptr, sizeof(sgx_key_t)) != 0) ||
-            (memcmp(key_buffer, third_key_buffer, sizeof(sgx_key_t)) != 0))
+        if ((memcmp(key_buffer_ptr, third_key_ptr, sizeof(sgx_key_t)) != 0))
         {
             return false;
```

Comment:
I think you need to free third_key_ptr before returning false

Reply:
Fixed